### ISSUE TITLE
Implement registration functionality

### ISSUE THREAD

**Issue by ndy40** [ndy40/bookmarking-app#8]
We currently have a static registration page with minimal functionality. Flesh out the rest of the workflow for registration. 

Acceptance Criteria:
- Form validation works and we get the required functionality. 
- Once registration is complete, close form and show a success message. 
- Make sure registration email is sent with account activation link. The link should expire in 2 days
- Make sure that duplicate registration is not allowed. Email is the unique key here.

**Comment by ndy40** [comment]
Feature implemented and merged into develop
